polling_stations/apps/data_finder/helpers/baked_data_helper.py

```diff
@@ -0,0 +1,170 @@
+import abc
+import enum
+from pathlib import Path
+from typing import List, Optional, Tuple, Type
+from urllib.parse import urljoin
+
+import polars
+from django.conf import settings
+from requests import Session
+from uk_geo_utils.helpers import Postcode
+
+session = Session()
+
+
+@enum.unique
+class BakedElectionsHelperMode(enum.Enum):
+    REMOTE = "REMOTE"
+    S3 = "S3"
+    LOCAL = "LOCAL"
+
+
+def ballot_paper_id_to_static_url(ballot_paper_id):
+    parts = ballot_paper_id.split(".")
+    path = "/".join((parts[-1], parts[0], parts[1], f"{ballot_paper_id}.json"))
+    return urljoin(settings.WCIVF_BALLOT_CACHE_URL, path)
+
+
+class BaseBakedElectionsStrategy(abc.ABC):
+    def __init__(self, **kwargs):
+        ...
+
+    @abc.abstractmethod
+    def get_response_for_postcode(self, postcode: Postcode):
+        raise NotImplementedError
+
+
+class RemoteBakedElectionsStrategy(BaseBakedElectionsStrategy):
+    def __init__(self, api_key=None, **kwargs):
+        self.api_key = api_key or getattr(settings, "DEVS_DC_API_KEY", None)
+        if not self.api_key:
+            raise ValueError("API key required for remote backend")
+        super().__init__(**kwargs)
+
+    def get_response_for_postcode(self, postcode: Postcode):
+        # TODO: pull into a constant
+        devs_dc_base = "https://developers.democracyclub.org.uk"
```

Comment:
Want to do this now? I don't mind.

Reply:
Done, although the `RemoteBakedElectionsHelper` isn't actually uses anywhere at the moment. I'll keep it in for now, but we might never need it